_posts/en/posts/2016-02-26-zkcp.md

```diff
@@ -0,0 +1,140 @@
+author: gmaxwell
+tags: [privacy]
+version: 1
+excerpt: Announcement for the first Zero-Knowledge Contingent Payment on the Bitcoin Network.
+---
+I am happy to announce the first successful  Zero-Knowledge Contingent Payment (ZKCP) on the Bitcoin Network.
+
+ZKCP is a transaction protocol that allows a buyer to purchase information from a seller using Bitcoin in a manner which is private, scalable, secure, and which doesn't require trusting anyone: the expected information is transferred if and _only_ if the payment is made. The buyer and seller do not need to trust each other or depend on arbitration by a third party.
+
+Imagine a movie-style "briefcase swap" (one party with a briefcase full of cash, another containing secret documents), but without the potential scenario of one of the cases being filled with shredded newspaper and the resulting exciting chase scene.
+
+An example application would be the owners of a particular make of e-book reader cooperating to purchase the DRM master keys from a failing manufacturer, so that they could load their own documents on their readers after the vendor's servers go offline. This type of sale is inherently
+irreversible, potentially crosses multiple jurisdictions, and involves parties whose financial stability is uncertain--meaning that both parties either take a great deal of risk or have to make difficult arrangement. Using a ZKCP avoids the significant transactional costs involved in a
+sale which can otherwise easily go wrong. 
+
+In today's transaction I purchased a solution to a 16x16 Sudoku puzzle for 0.10 BTC  from Sean Bowe, a member of the Zcash team, as part of a demonstration performed live at Financial Cryptography 2016 in Barbados. I played my part in the transaction remotely from California.
+
+The transfer involved two transactions: 
+
+- [FIXME](https://www.blocktrail.com/BTC/tx/FIXME)
+- [FIXME](https://www.blocktrail.com/BTC/tx/FIXME)
+
+Almost all of the engineering work behind this ZKCP implementation was done by Sean Bowe, with support from Pieter Wuille, myself, and Madars Virza.
+
+## Background
+
+I first proposed the ZKCP protocol in 2011 in an [article on the Bitcoin Wiki](https://en.bitcoin.it/wiki/Zero_Knowledge_Contingent_Payment) as an example of how tremendously powerful the existing primitives in Bitcoin Script already were.
+
+### Zero Knowledge Proofs
+
+My ZKCP protocol required as a building block a zero-knowledge proof for arbitrary programs. Many kind of specialized zero-knowledge proofs exist:
+common digital signatures are an example, as are the range proofs in [Confidential Transaction](https://people.xiph.org/~greg/confidential_values.txt).
+
+A zero-knowledge proof for _general computation_ is a cryptographic system which lets a person run an arbitrary program with a mixture of public and secret inputs and prove to others that this specific program accepted the inputs, without revealing anything more about its operation or the secret inputs.
```

Comment:
Would it be appropriate to emphasize here that "acceptance" means "faithful execution and some computational result"? "Accepting the inputs" otherwise means they were recognized as inputs, right?